dlt/destinations/impl/duckdb/configuration.py

```diff
@@ -226,10 +211,18 @@ def __init__(
         create_indexes: bool = False,
         destination_name: str = None,
         environment: str = None,
+        bound_to_pipeline: Optional[SupportsPipeline] = None,
     ) -> None:
         super().__init__(
             credentials=credentials,  # type: ignore[arg-type]
             destination_name=destination_name,
             environment=environment,
         )
+        self.bound_to_pipeline = bound_to_pipeline
         self.create_indexes = create_indexes
+
+    def on_resolved(self) -> None:
+        # pass bound pipeline to duckdb credentials
+        # TODO: find a better way to pass and bind explicit pipeline context
```

Comment:
I am also not sure, but maybe we should not have an "active pipeline" but rather some pipeline context object that is shared among the whole object tree of a given pipeline, similar to how other libraries do it. But I'm not sure what other consequences would come from not having active pipelines anymore. Also what happens if we actually start using dataset objects as "destinations" and share them between multiple pipelines, do we then have to replace this context object everytime? In this scenario the active pipeline probably is the better so solution..

Reply:
the primary use of active pipeline is to have access to state, schemas, clients etc. in dlt source and resources. this is OK because we execute them from within the pipeline (except sources that may be instantiated before pipeline is created and activated, cost of freedom ;>)